ultraplot/config.py

```diff
@@ -1797,6 +1815,20 @@ def save(self, path=None, user=True, comment=None, backup=True, description=Fals
         user_dict = self.changed if user else None
         self._save_yaml(path, user_dict, comment=comment, description=description)
 
+    def sync(self):
+        """
+        Sync settings between ultraplot and matplotlib rc dictionaries.
+        """
+        for _src in (self.rc_ultraplot, self.rc_matplotlib):
+            for _key in _src:  # loop through unsynced properties
+                if "color" not in _key:
+                    continue
+                try:
+                    _src[_key] = _src[_key]
+                except ValueError as err:
+                    warnings._warn_ultraplot(f"Invalid user rc file setting: {err}")
+                    _src[_key] = "black"  # fill value
```

Comment:
I don't follow how this code works to be honest. Also the method should probably be called `sync_colors` since it only deals with colors?

Reply:
> I don't follow how this code works to be honest. Also the method should probably be called `sync_colors` since it only deals with colors?

actually taht is a good point -- will change it for clarity

Reply:
At least for the `rc_ultraplot` it will run a validator on setting a key (see `rcsetup.py`); matplotlib prolly has a similar setting.

Initially I was also scratching my head at this section in the code.  Will replace it as we progress to something a bit cleaner.